### PR TITLE
Add username

### PR DESCRIPTION
**Pull request by patrick-rebel**
I AddUsernameToUsers, update sign up form, users controller, application.html.erb

### CODE REVIEW

**Comment by patrick-rebel** [comment]
Next time I will write the test first haha. I'm learning every day!

**Comment by antwonlee** [comment]
@patrick-rebel Perhaps, you can use factories and create a user there?

**Comment by patrick-rebel** [comment]
I don't know if I must add the fields full_name, last_name, bio to the registration form can someone look at this?

**Comment by antwonlee** [comment]
@patrick-rebel Here are my thoughts:

- Rather than doing a feature test to check bio's length, just do that in the model test with shoulda matchers.  Saying that, I don't think it is necessary that bio needs to be a required field.  I think they just wanted to make sure if there is bio, it doesn't exceed to certain character limit.

- In your feature test, I think it is cleaner to put the values in the factories and following this template:

test do
  setup
  exercise
  verify
  teardown
end

Here is Chris's note on good feature spec: https://github.com/ajkamel/cover/blob/ec1487fa547f8bce071ac909de18797201ca67b1/spec/features/user_views_all_books_spec.rb#L4-L12

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -19,6 +19,11 @@ def create
   private
 
   def user_params
-    params.require(:user).permit(:email, :password)
+    params.require(:user).permit(:username,
+                                 :email,
+                                 :password,
+                                 :first_name,
+                                 :last_name,
+                                 :bio)
```
Comment (antwonlee) [inline app/controllers/users_controller.rb]: Maybe we can do indentation differently here.

File: db/migrate/20160127124620_add_username_to_users.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class AddUsernameToUsers < ActiveRecord::Migration
+  def change
+    add_column :users, :username, :string
```
Comment (antwonlee) [inline db/migrate/20160127124620_add_username_to_users.rb]: I think we can make this `unique: true`

**Pull request closed without merging**